### PR TITLE
Beacon api: fix get blind block

### LINKED ISSUE #11293: Blinded block generation should not require validator registration
# 🐞 Bug Report

### Description

When attempting to generate a blinded beacon block prysm currently refuses to do so unless the validator has been registered.  However, in configurations with multiple beacon nodes or alternative registration systems the beacon node may not have been informed of the registration.  As such, it should not check for local knowledge of the registration as a gate to fetching blocks from the MEV relay.

Relevant code: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go#L45-L46

It may be desirable to carry out a check that the returned block header has a non-zero fee recipient for situations where the relay really does not know the fee recipient address. 

### Has this worked before in a previous version?

No.

## 🌍  Your Environment

Prysm v3.0.0

### PR DESCRIPTION
**Pull request by terencechain**
Fixes #11293

The current beacon API for getting blind blocks has many incorrect assumptions:
- Returns blocks from all forks 
- The validator has to be registered through the same beacon node

Instead, we should simplify the get blind block endpoint to return an error if any of these is true
 - The current time is before Bellatrix fork epoch.
 - The requested slot is before Bellatrix fork epoch.
 - The node is the syncing or optimistic mode.
 - The builder is not figured.
 - The relayer circuit breaker is activated.
 - The relayer responded with an error.

In the happy path, the endpoint will return a block with a header that's through the relay/builder network

### CODE REVIEW

File: beacon-chain/rpc/eth/validator/validator.go [hunk 1]
```diff
@@ -404,66 +404,65 @@ func (vs *Server) ProduceBlockV2SSZ(ctx context.Context, req *ethpbv1.ProduceBlo
 // ProduceBlindedBlock requests the beacon node to produce a valid unsigned blinded beacon block,
 // which can then be signed by a proposer and submitted.
 //
-// Pre-Bellatrix, this endpoint will return a regular block.
+// Under the following conditions, this endpoint will return an error.
+// - The current time is before bellatrix fork epoch.
+// - The requested slot is before bellatrix fork epoch.
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:409]: This is against the spec: https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/publishBlindedBlock
> Pre-Bellatrix, this endpoint will accept a SignedBeaconBlock.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: beacon-chain/rpc/eth/validator/validator.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/rpc/eth/validator/validator.go:409]: That's publish, not produce. but i see it's the same with produce, that's annoying, I'll have to add the original code back: 
https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/produceBlindedBlock

Comment (terencechain) [reply beacon-chain/rpc/eth/validator/validator.go:409]: It makes no sense to produce a non-blind block IMO

File: beacon-chain/rpc/eth/validator/validator.go [hunk 2]
```diff
@@ -413,57 +417,76 @@ func (vs *Server) ProduceBlindedBlock(ctx context.Context, req *ethpbv1.ProduceB
...
+
+	// After Bellatrix, return blinded block.
+	optimistic, err := vs.OptimisticModeFetcher.IsOptimistic(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not determine if the node is a optimistic node: %v", err)
+	}
+	if optimistic {
+		return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
+	}
+	altairBlk, err := vs.V1Alpha1Server.BuildAltairBeaconBlock(ctx, v1alpha1req)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:469]: Why do we build an Altair block here, not Bellatrix?

Comment (terencechain) [reply beacon-chain/rpc/eth/validator/validator.go:469]: Build Altair block as the base, then insert header to turn that into a blind Bellatrix block.
Similar workflow for the non-blind block as well

**Review by rkapka — APPROVED** [review]

**Pull request merged**